runtime/compiler/control/CompilationThread.cpp

```diff
@@ -3778,6 +3779,26 @@ void TR::CompilationInfo::stopCompilationThreads()
          }
       }
 #endif /* defined(J9VM_OPT_JITSERVER) */
+   // check if there is any compilation logged in the global list, if so, dump the contents
+   if (J9::Options::_collectRegionLog && J9::Options::_compilationRegionLogFileName)
+      {
+      std::FILE *regionLogOutFile = fopen(J9::Options::_compilationRegionLogFileName, "w+");
+      if (!regionLogOutFile)
+         {
+         fprintf(stderr, "error in opening file %s for dumping region logs\n", J9::Options::_compilationRegionLogFileName);
+         return;
+         }
+      // TR::Region::printAllCompilations(J9::SystemSegmentProvider::_globalCompilationsList, regionLogOutFile);
+      if (J9::SystemSegmentProvider::_globalCompilationsList)
+         {
+         for (auto &compilation : *J9::SystemSegmentProvider::_globalCompilationsList)
+            {
+            fprintf(regionLogOutFile, "Compilation %d, Memory usage: %zu\n", std::get<0>(compilation), std::get<1>(compilation));
```

Comment:
Do you print the name of the method being compiled and the optimization level anywhere?

Reply:
Emmm, that'sanother of my concern, I plan to make the thing a struct instead of the tuple and include these information, but I don't know how to access these fields.

Reply:
Through the compilation queue entry ("method being compiled") when the segment provider is created. I thought I showed you that at some point? I think you can get the "IL generator method details" or something like that from the method being compiled, and the `J9Method` from there, then `J9ROMMethod` and `J9ROMClass` from there, and eventually the class name and the method name and signature. Let me know if you have trouble figuring this out and I'll give you the actual details.

Reply:
Oh that complicated and lengthy way of getting it. I fear it's overwritten by newly added code.

Reply:
All the APIs involved in doing this are supposed to be pretty stable, so I don't think that's a concern. I think it's important to include the name and the optimization level of the compiled method in the output.